### PR TITLE
Account sign in/out changes

### PR DESCRIPTION
**Pull request by maxme**

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java [hunk 1]
```diff
@@ -126,12 +126,12 @@ public void onClick(View view) {
         }
     }
 
-    private void signoutWithConfirmation() {
+    private void signOutWordPressComWithConfirmation() {
         new AlertDialog.Builder(getActivity())
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java]: This confirmation should be different (or there should be an additional confirmation) if signing out will remove local drafts.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 2]
```diff
@@ -375,7 +376,7 @@ private boolean isViewingNotificationsTab() {
     // Events
 
     @SuppressWarnings("unused")
-    public void onEventMainThread(CoreEvents.UserSignedOut event) {
+    public void onEventMainThread(UserSignedOutWordPressCom event) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: I think this should be changed to handle `UserSignedOut` separately, like this:
```
@SuppressWarnings("unused")
public void onEventMainThread(UserSignedOutWordPressCom event) {
    resetFragments();
}

@SuppressWarnings("unused")
public void onEventMainThread(CoreEvents.UserSignedOut event) {
    ActivityLauncher.showSignInForResult(this);
}
```
Also, `UserSignedOut` would be more clear if it was renamed `UserSignedOutCompletely`.

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: Fixed in b662a73

**Comment by maxme** [comment]
I reused the wpios sign out message + specified local drafts and changes will be deleted.

**Comment by maxme** [comment]
Reopen https://github.com/wordpress-mobile/WordPress-Android/issues/1989 on merge

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**